cypress/e2e/webTables.cy.js

```diff
@@ -1,7 +1,159 @@
-/// <reference types='cypress' />
+describe("Web Tables page", () => {
+  let user;
 
-describe('Web Tables page', () => {
-  it('', () => {
+  beforeEach(() => {
+    cy.task('generateUser').then((generateUser) => {
+      user = generateUser;
+    });
+    cy.visit("https://demoqa.com/webtables");
+  });
+
+  it("should have the pagination", () => {
+    cy.get(".-pagination").should("exist");
+    cy.get(".-pagination").should("contain.text", "Previous");
+    cy.get(".-pagination").should("contain.text", "Page");
+    cy.get(".-pagination").should("contain.text", "Next");
+    cy.get(".-pagination .-pageJump input").should("have.attr", "type", "number");
+  });
+
+  it("should have Rows count selection", () => {
+    cy.get('[aria-label="rows per page"]').should("exist");
+    cy.get('[aria-label="rows per page"]').select("5 rows").should('have.value', '5');
+    cy.get('[aria-label="rows per page"]').select("10 rows").should('have.value', '10');
+    cy.get('[aria-label="rows per page"]').select("20 rows").should('have.value', '20');
+    cy.get('[aria-label="rows per page"]').select("25 rows").should('have.value', '25');
+  });
+
+  it("should add a new worker", () => {
+    for (let i = 0; i < 2; i++) {
+      cy.log(`Iteration ${i + 1}`);
+      cy.get("#addNewRecordButton").click();
+      cy.get("#firstName").type(user.firstName);
+      cy.get("#lastName").type(user.lastName);
+      cy.get("#userEmail").type(user.email);
+      cy.get("#age").type(user.age);
+      cy.get("#salary").type(user.salary);
+      cy.get("#department").type(user.department);
+      cy.get("#submit").click();
+      cy.get(".rt-tbody").should("contain", user.firstName);
+      cy.get(".rt-tbody").should("contain", user.lastName);
+      cy.get(".rt-tbody").should("contain", user.email);
+      cy.get(".rt-tbody").should("contain", user.age);
+      cy.get(".rt-tbody").should("contain", user.salary);
+      cy.get(".rt-tbody").should("contain", user.department);
+    }
+  });
+
+  it("should validate data in worker row after creating worker", () => {
```

Comment:
what is the difference between this test and previous?

Reply:
can i merge 2 in 1 test cases ?